src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1924,6 +1928,13 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 
 	workerExtendedOpNode = CitusMakeNode(MultiExtendedOp);
 	workerExtendedOpNode->targetList = newTargetEntryList;
+
+	if (!queryHasAggregates && originalOpNode->distinctClause)
```

Comment:
1. If this block returns false, `workerExtendedOpNode->distinctClause` and `workerExtendedOpNode->hasDistinctOn` stays uninitialized.
2. Do we need the `if` block, can't we just assign values from `originalOpNode`? What is the problem with `queryHasAggregates`?
3. Do we need `copyObject ` for copyObject?

Reply:
1. they are initialized to NIL, and false respectively by ```CitusMakeNode()``` few lines above.
2. correct.  It has the same effect. 
3. we can get by without copyObject, removing.